spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java

```diff
@@ -506,19 +508,22 @@ else if (this.micrometerEnabled) {
 		}
 	}
 
+	private String removeLeadingAndTrailingBrackets(String str) {
```

Comment:
This should be `static` and somewhere in the end of class.
Probably also no need to make its code multi-line: looks like that last `']'` would fit into line length.